canonical-web-and-design/ubuntu.com
# Add CUBE dashboard POC

## Done

- Fetch enrollments data from Open edX
  - Display daily and monthly enrollments in a chart filterable by course
- Fetch exam data from Open edX and display it in a table
- CSV download button for enrollments and exam attempts

## QA

- Run the server with dotrun
- Go to http://localhost:8001/cube/dashboard (or http://localhost:8001/cube/dashboard?test_backend=true to see QA data)
- Enrollment data should be displayed in a stacked bar chart
  - Use the search and filter component to filter by course
  - Use the select component to show monthly enrollments
- Enrollment data should be displayed in a table
- Exam attempt data should be displayed in a table

## Issue / Card

https://github.com/canonical-web-and-design/commercial-squad/issues/360
https://github.com/canonical-web-and-design/commercial-squad/issues/402

[Notes](https://docs.google.com/document/d/1SFW9eyrK3oSu38Mde5x1f6hKWWv0hWe_k0kdlH7aOL8/edit)

## Screenshots

![cube-dashboard-monthly](https://user-images.githubusercontent.com/995051/142175875-cd9a8a50-d79b-4ecc-95ab-5011dc90100e.png)

![cube-dashboard-daily](https://user-images.githubusercontent.com/995051/142175887-b5abb435-6d80-4913-b8ef-0cccc2e32636.png)
